metrics_utility/automation_controller_billing/report/report_renewal_guidance.py

```diff
@@ -157,8 +157,12 @@ def get_intervals(self, start_date, end_date, interval_size):
 
     def compute_ephemeral_intervals(self, host_metric_dataframe):
         # Convert input date strings to datetime objects
-        start_date = pd.to_datetime(self.extra_params['since_date']).tz_localize(None)
-        end_date = pd.to_datetime(self.extra_params['until_date']).tz_localize(None) + datetime.timedelta(days=1) - datetime.timedelta(microseconds=1)
+        start_date = pd.to_datetime(self.extra_params['since_date'], format='ISO8601').tz_localize(None)
+        end_date = (
+            pd.to_datetime(self.extra_params['until_date'], format='ISO8601').tz_localize(None)
```

Comment:
hm I wonder if this can break some customer integrations. Customers ere exporting this data into visualizations tools like Power BI. Technically it should handle several datetime formats I think